### PR TITLE
Added object enum for user status.

### PR DESCRIPTION
**Pull request by SanketDhabarde**
### Changes made
Converted:
```js
const userStatusEnum = ["ooo", "idle", "active"];
```
To:
```js
const USER_STATUS = {
  OOO: "ooo",
  IDLE: "idle",
  ACTIVE: "active",
};
```
And used it at the places where `userStatusEnum` was used

### Dependent tasks
- https://github.com/Real-Dev-Squad/website-backend/issues/754
- https://github.com/Real-Dev-Squad/website-backend/issues/766

### CODE REVIEW

File: controllers/tasks.js [hunk 1]
```diff
@@ -211,7 +212,7 @@ const assignTask = async (req, res) => {
   try {
     const { status, username } = req.userData;
 
-    if (status !== userStatusEnum[1]) {
+    if (status !== IDLE) {
```
Comment (samarpan1738) [inline controllers/tasks.js:215]: This would be more readable
```suggestion
    if (status !== userStatus.IDLE) {
```

Comment (SanketDhabarde) [reply controllers/tasks.js:215]: No, There is a better way. If you want to use more status suppose "active", you will do it like `userStatus.ACTIVE`. and for "ooo" you will use `userStatus.OOO`. So if we use an object to access property more than 2 times, we usually destructure the object.

File: constants/users.js [hunk 2]
```diff
@@ -1,7 +1,13 @@
-const userStatusEnum = ["ooo", "idle", "active"];
 const profileStatus = {
   PENDING: "PENDING",
   APPROVED: "APPROVED",
   NOT_APPROVED: "NOT APPROVED",
 };
-module.exports = { userStatusEnum, profileStatus };
+
+const userStatus = {
```
Comment (samarpan1738) [inline constants/users.js:7]: Constants are usually uppercase
```suggestion
const USER_STATUS = {
```

File: controllers/tasks.js (hunk 1 above)
Comment (samarpan1738) [reply controllers/tasks.js:215]: In RDS we task and user both have ACTIVE status. If we destructure, it'll be ambiguous that which entity the status belongs

Comment (SanketDhabarde) [reply controllers/tasks.js:215]: In that case. it makes sense not to destructure. I will make the change thank you.

**Review by samarpan1738 — APPROVED** [review]
LGTM

**Review by vinayak-trivedi — APPROVED** [review]
LGTM 👍

**Pull request merged**